### PR TITLE
Improve readability of `adsConfig` logic for consent properties `nonPersonalizedAd ` and `restrictedDataProcessor `

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

Improve readability of `adsConfig` logic for consent properties `nonPersonalizedAd ` and `restrictedDataProcessor `.

Convert to TypeScript.

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### Tested

- [ ] Locally
- [ ] On CODE (optional)

### CODE REVIEW

**Review by mxdvl — APPROVED** [review]
LGTM, but are we sure this also covers AUS?

**Comment by arelra** [comment]
> LGTM, but are we sure this also covers AUS?

Yeah aus is strange here because looking at `getConsentChange` we seem to cover both aus and ccpa with ccpa.

**Comment by mxdvl** [comment]
Do you mean `getConsentFor` or `onConsentChange`?

**Comment by arelra** [comment]
> Do you mean `getConsentFor` or `onConsentChange`?

Sorry its [onConsentChange](https://github.com/guardian/frontend/blob/75fe4a938876f242c675d532333f8941bc2e9d6e/static/src/javascripts/projects/common/modules/atoms/youtube-player.js#L38)

**Review by mxdvl — APPROVED** [review]
Obviously, it’d be great to turn this to TS.

Excellent simplification here. Just a small comment to say `framework` might be more accurate that `region`.

**Review by mxdvl — APPROVED** [review]
oh-em-gee, you actually TS’d it <3 <3 +++

File: static/src/javascripts/projects/common/modules/atoms/youtube-player.ts [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+interface ConsentState {
+	framework: null | string;
+	canTarget: boolean;
+}
+
+let consentState: ConsentState = {
+	framework: null,
+	canTarget: false,
+};
```
Comment (MarSavar) [inline static/src/javascripts/projects/common/modules/atoms/youtube-player.ts:47]: Thank you so much for doing this. ❤️ 
Having a `consentState` object which is framework-agnostic is huge improvement.
`canTarget` is also especially intuitive and meaningful (as opposed to `ccpaState` and `tcfState`).

**Pull request merged**